torchgeo/trainers/autoregression.py

```diff
@@ -0,0 +1,201 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""Trainers for autoregression."""
+
+from typing import Any
+
+import torch.nn as nn
+from torch import Tensor
+from torchmetrics import MetricCollection
+from torchmetrics.regression import MeanAbsoluteError, MeanSquaredError
+
+from torchgeo.models import LSTMSeq2Seq
+
+from .base import BaseTask
+
+
+class AutoregressionTask(BaseTask):
+    """Autoregression."""
+
+    def __init__(
+        self,
+        model: str = 'lstm_seq2seq',
```

Comment:
```suggestion
        model: Literal['seq2seq'] = 'seq2seq',
        cell: Literal['rnn', 'gru', 'lstm'] = 'lstm',
```
or something like that